server/utils/hollaex-tools-lib/tools/security.js

```diff
@@ -1029,13 +1037,18 @@ const createUserKitHmacToken = async (userId, otpCode, ip, name, role = ROLES.US
 async function updateUserKitHmacToken(userId, otpCode, ip, token_id, name, permissions, whitelisted_ips, whitelisting_enabled) {
 	await checkUserOtpActive(userId, otpCode);
 	const token = await findToken({ where: { id: token_id } });
+	const user = await getModel('user').findOne({ where: { id: userId } });
 
 	if (!token) {
 		throw new Error(TOKEN_NOT_FOUND);
 	} else if (token.revoked) {
 		throw new Error(TOKEN_REVOKED);
 	}
 
+	if(!whitelisting_enabled && user.is_admin) {
```

Comment:
You should check the token role to be "admin" instead of getting the user and checking if thats admin. there is no need to get user here @fetok12